content/en/blog/_posts/2018-09-19-network-bootable-farm-with-ltsp.md

```diff
@@ -0,0 +1,485 @@
+    RCFILE_03_KERNEL_DEBUG         = "sysctl -w kernel.unknown_nmi_panic=1 kernel.softlockup_panic=1; modprobe netconsole netconsole=@/vmbr0,@10.9.0.15/"
+    RCFILE_04_SYSCTL               = "sysctl -w fs.file-max=20000000 fs.nr_open=20000000 net.ipv4.neigh.default.gc_thresh1=80000 net.ipv4.neigh.default.gc_thresh2=90000 net.ipv4.neigh.default.gc_thresh3=100000"
+    RCFILE_05_FORWARD              = "echo 1 > /proc/sys/net/ipv4/ip_forward"
+    RCFILE_06_MODULES              = "modprobe br_netfilter"
+    RCFILE_07_JOIN_K8S             = "kubeadm join --token 2a4576.504356e45fa3d365 10.9.0.20:6443 --discovery-token-ca-cert-hash sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
+```
+
+* **KEEP_SYSTEM_SERVICES** - during the boot LTSP automatically removes some services, this variable is needed for prevent this behavior.
+* **PREINIT_*** - commands listed here will be executed before systemd running (this functional was added by [feature_preinit.diff](#used-patches-list) patch)
+* **FSTAB_*** - entries writed here will be added to `/etc/fstab` file.
+As you can see, I use `nofail` option, that means that if partition not existing, it will continue boot without error.
+* **RCFILE_*** - those commands will be writed to `rc.local` file, which will be called by systemd during the boot.
+Here I load kernel modules and add  some sysctl tunes, then call `kubeadm join` command, which adds my node to the kubernetes cluster.
+
+More details about all variables used here you can get from [lts.conf manpage](http://manpages.ubuntu.com/manpages/xenial/man5/lts.conf.5.html).
+
+Now you can configure your DHCP, basically you should set `next-server` and `filename` options.
+
+I use ISC-DHCP server, and here is example `dhcpd.conf`:
+
+```
+shared-network ltsp-netowrk {
+    subnet 10.9.0.0 netmask 255.255.0.0 {
+        authoritative;
+        default-lease-time -1;
+        max-lease-time -1;
+
+        option domain-name              "example.org";
+        option domain-name-servers      10.9.0.1;
+        option routers                  10.9.0.1;
+        next-server                     ltsp-1;  # write ltsp-server hostname here
+
+        if option architecture = 00:07 {
+            filename "/ltsp/amd64/grub/x86_64-efi/core.efi";
+        } else {
+            filename "/ltsp/amd64/grub/i386-pc/core.0";
+        }
+
+        range 10.9.200.0 10.9.250.254; 
+    }
+```
+
+You can start from this, but what about me, I have multiple ltsp-servers and I configure leases statically for each node via ansible playbook.
+
+Try to run your first node if everything was right, you will have running system there.
+Node also will be added to your Kubernetes cluster.
+
+Now you can try to do your own changes. As I said LTSP is very easy customizable, so here is shouldn't be anything difficults.
```

Comment:
possibly the sentence needs a little change?

> As I said, LTSP is very easy customizable, so you shouldn't have any difficulties.

if i understand the context correctly.

might as well just leave:
> Now you can try to do your own changes.

Reply:
Changed to:

>If you need something more, note that LTSP can be easily changed to your needs.
>Feel free to look into source code you can find many answers there.